File: mobile/lib/features/services/data/catalog_models.dart
class ServiceCategory {
  const ServiceCategory({
    required this.id,
    required this.name,
    this.description,
    this.imageUrl,
    required this.isActive,
    required this.serviceCount,
  });

  factory ServiceCategory.fromJson(Map<String, dynamic> json) {
    return ServiceCategory(
      id: json['id'] as String,
      name: json['name'] as String,
      description: json['description'] as String?,
      imageUrl: json['imageUrl'] as String?,
      isActive: json['isActive'] as bool? ?? true,
      serviceCount: json['serviceCount'] as int? ?? 0,
    );
  }

  final String id;
  final String name;
  final String? description;
  final String? imageUrl;
  final bool isActive;
  final int serviceCount;
}

class CatalogService {
  const CatalogService({
    required this.id,
    required this.categoryId,
    required this.categoryName,
    required this.name,
    this.description,
    required this.basePrice,
    required this.isActive,
    this.estimatedDurationMinutes,
  });

  factory CatalogService.fromJson(Map<String, dynamic> json) {
    return CatalogService(
      id: json['id'] as String,
      categoryId: json['categoryId'] as String,
      categoryName: json['categoryName'] as String? ?? '',
      name: json['name'] as String,
      description: json['description'] as String?,
      basePrice: (json['basePrice'] as num).toDouble(),
      isActive: json['isActive'] as bool? ?? true,
      estimatedDurationMinutes: json['estimatedDurationMinutes'] as int?,
    );
  }

  final String id;
  final String categoryId;
  final String categoryName;
  final String name;
  final String? description;
  final double basePrice;
  final bool isActive;
  final int? estimatedDurationMinutes;
}

File: admin-web/src/app/shared/models/catalog.models.ts
export interface ServiceCategory {
  id: string;
  name: string;
  description?: string | null;
  imageUrl?: string | null;
  isActive: boolean;
  serviceCount: number;
  createdAt: string;
  updatedAt?: string | null;
}

export interface ServiceCategoryPayload {
  name: string;
  description?: string | null;
  imageUrl?: string | null;
  isActive: boolean;
}

export interface CatalogService {
  id: string;
  categoryId: string;
  categoryName: string;
  name: string;
  description?: string | null;
  basePrice: number;
  isActive: boolean;
  estimatedDurationMinutes?: number | null;
  createdAt: string;
  updatedAt?: string | null;
}

export interface CatalogServicePayload {
  categoryId: string;
  name: string;
  description?: string | null;
  basePrice: number;
  estimatedDurationMinutes?: number | null;
  isActive: boolean;
}

export interface ServiceFilters {
  categoryId?: string | null;
  search?: string | null;
  includeInactive?: boolean;
}
